ObservatoryHerald/SpeechRequestManager.cs

```diff
@@ -0,0 +1,284 @@
+﻿using Observatory.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Xml;
+using System.Security.Cryptography;
+
+namespace Observatory.Herald
+{
+    class SpeechRequestManager
+    {
+        private HttpClient httpClient;
+        private string ApiKey;
+        private string ApiEndpoint;
+        private DirectoryInfo cacheLocation;
+        private int cacheSize;
+        
+        internal SpeechRequestManager(HeraldSettings settings, HttpClient httpClient)
+        {
+            ApiKey = ObservatoryAPI.ApiKey;
+            ApiEndpoint = settings.ApiEndpoint;
+            this.httpClient = httpClient;
+            cacheSize = settings.CacheSize;
+            cacheLocation = new(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
+                + $"{Path.DirectorySeparatorChar}ObservatoryCore{Path.DirectorySeparatorChar}ObservatoryHerald{Path.DirectorySeparatorChar}");
+            
+            if (!Directory.Exists(cacheLocation.FullName))
+            {
+                Directory.CreateDirectory(cacheLocation.FullName);
+            }
+
+            settings.Voices = PopulateVoiceSettingOptions();
+        }
+
+        internal string GetAudioFileFromSsml(string ssml, string voice, string style, string rate)
+        {
+
+            ssml = AddVoiceToSsml(ssml, voice, style, rate);
+
+            using var sha = SHA256.Create();
+
+            var ssmlHash = BitConverter.ToString(
+                sha.ComputeHash(Encoding.UTF8.GetBytes(ssml))
+                ).Replace("-", string.Empty);
+
+            var audioFilename = cacheLocation + ssmlHash + ".mp3";
+
+            if (!File.Exists(audioFilename))
+            {
+                using StringContent request = new(ssml)
+                {
+                    Headers = {
+                        { "obs-plugin", "herald" },
+                        { "api-key", ApiKey }
+                    }
+                };
+                
+                using var response = httpClient.PostAsync(ApiEndpoint + "/Speak", request).Result;
+
+                if (response.IsSuccessStatusCode)
+                {
+                    using FileStream fileStream = new FileStream(audioFilename, FileMode.CreateNew);
+                    response.Content.ReadAsStream().CopyTo(fileStream);
+                    fileStream.Close();
```

Comment:
No personal reason, and had actually done so originally, but Intellisense specifically suggested not doing so, and I'm willing to believe that Microsoft's engineers have a better sense of how using interacts with block scoping than I do. ;)

Reply:
Looking again, I suspect it's because there's nothing else in this if block that happens after `fileStream` would be disposed before the block goes out of scope, so block form and line form are synonymous in this situation.

Reply:
Oh, no it's not disposed yet. The declaration syntax for `using` disposes when the object goes out of scope, not immediately after the next line.